monai/metrics/metric.py

```diff
@@ -15,121 +15,165 @@
-        And users can execute on every batch of data, then accumulate the results, or
-        accumulate the original `y_pred` and `y`, then execute on the accumulated data.
+        Execute basic computation for model prediction `y_pred` and ground truth `y` (optional).
+        It supports inputs of a list of "channel-first" Tensor and a "batch-first" Tensor.
 
         Args:
-            y_pred: the model prediction data to compute, must be a list of `channel-first` Tensor
+            y_pred: the raw model prediction data at one iteration, must be a list of `channel-first` Tensor
                 or a `batch-first` Tensor.
             y: the ground truth to compute, must be a list of `channel-first` Tensor
                 or a `batch-first` Tensor.
 
+        Returns:
+            The computed metric values at the iteration level.
+            The output shape could be a `batch-first` tensor or a list of `batch-first` tensors.
+            When it's a list of tensors, each item in the list can represent (batched) class-wise metrics.
+
         """
         ret: TensorOrList
+        # handling a list of channel-first data
         if isinstance(y_pred, (list, tuple)) or isinstance(y, (list, tuple)):
-            # if y_pred or y is a list of channel-first data, add batch dim and compute metric
-            ret = self._compute_list(y_pred, y)
-        elif isinstance(y_pred, torch.Tensor):
-            y_ = y.detach() if y is not None and isinstance(y, torch.Tensor) else None
-            ret = self._compute_tensor(y_pred.detach(), y_)
-        else:
-            raise ValueError("y_pred or y must be a list of `channel-first` Tensors or a `batch-first` Tensor.")
-
-        return ret
+            return self._compute_list(y_pred, y)
+        # handling a single batch-first data
+        if isinstance(y_pred, torch.Tensor):
+            y_ = None
+            if y is not None and isinstance(y, torch.Tensor):
+                y_ = y.detach()
+            return self._compute_tensor(y_pred.detach(), y_)
+        raise ValueError("y_pred or y must be a list/tuple of `channel-first` Tensors or a `batch-first` Tensor.")
 
     def _compute_list(self, y_pred: TensorOrList, y: Optional[TensorOrList] = None):
         """
-        Excute the computation for the y_pred and y items of a iteration, the data is in the list shape.
-        Will concat the results to guarantee the output shape of ret is BCHW[D], otherwise it's list of batch-first,
-        which is against our principle that data in metrics should be BCHW[D] or list of channel-first.
-        Note: subclass may enhance the operation with multi-threads to accelerate.
+        Execute the metric computation for `y_pred` and `y` in a list of "channel-first" tensors.
 
+        The return value is a "batch-first" tensor, or a list of "batch-first" tensors.
+        When it's a list of tensors, each item in the list can represent (batched) class-wise metrics.
```

Comment:
I think usually the multi-class metrics is a multi-channel Tensor, not items of the list, here if every item of `ret` is a list, means the metric returns multiple items for every iteration, for example:
```py
class MyMetric:
    ...
    def _compute_tensor(self):
        # return some median computation result for every iteration, will aggregate for the final value later
        return TP, FP, WD
```

What do you think?

Thanks.